prefect_saturn/core.py

```diff
@@ -240,26 +233,29 @@ def add_environment(
                     "metadata": {"labels": {"identifier": ""}},
                     "spec": {
                         "restartPolicy": "Never",
-                        "hostAliases": host_aliases,
                         "containers": [
                             {
                                 "name": "flow-container",
                                 "image": "",
                                 "command": [],
                                 "args": [],
-                                "env": [{"name": k, "value": v} for k, v in job_env.items()],
-                                "envFrom": [{"secretRef": {"name": env_vars_secret_name}}],
+                                "env": [{"name": "BASE_URL", "value": self._base_url}],
```

Comment:
Now that I see what's left of this template, I think we should move the entire thing into Saturn's back-end. The only really prefect-specific things I see here are the `"identifier"` bits, and I feel pretty confident that we can remove those from our template and that Prefect will write in what it needs: https://github.com/PrefectHQ/prefect/blob/340b1b7ef6d1fdb12161b8964a19902659e4ff03/src/prefect/environments/execution/k8s/job.py#L280-L282.

Reply:
so then instead of "create part of a template, send it to Saturn, get it back with more stuff filled in" it would just be "given a flow's Saturn ID, ask Saturn for a template to use"

Reply:
even stuff not touched yet in this diff, like `flow_hash` and `job_name`, already exist in Saturn after calling `.register_flow_with_saturn()`, so we don't need to repeat them here

Reply:
Agreed. I changed this to no longer pass an incomplete job spec, and Saturn's API sending a full saturn-ified version back. Now it just requests a job spec for running flows, and uses that.